kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManagerImpl.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.controllers;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.kafka.serializers.subject.strategy.SubjectNameStrategy;
+import io.confluent.kafkarest.entities.EmbeddedFormat;
+import io.confluent.kafkarest.entities.RegisteredSchema;
+import java.io.IOException;
+import javax.inject.Inject;
+import org.apache.kafka.common.errors.SerializationException;
+
+final class SchemaManagerImpl implements SchemaManager {
+  private final SchemaRegistryClient schemaRegistryClient;
+  private final SubjectNameStrategy subjectNameStrategy;
+
+  @Inject
+  SchemaManagerImpl(
+      SchemaRegistryClient schemaRegistryClient, SubjectNameStrategy subjectNameStrategy) {
```

Comment:
for full generality, we'd want the ability to have more than one subject name strategy for keys or values i guess. you'd need a fancy way to specify that though, and RP would be the first place this requirement has come up. I think you could tack this in later if need be, so all good.

Reply:
Not as configuration though. So, my plan is that the configuration will only define the default strategy, in case you don't specify one, and let people specify the strategy per request. That will only be available in V3, of course. Currently, V2 does not even support specifying the one default strategy, so V2 is already winning here.

Reply:
yeah, seems a good plan 👍